contracts/core/Allo.sol

```diff
@@ -3,28 +3,52 @@ pragma solidity 0.8.19;
 
 import { Metadata } from "./libraries/Metadata.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "../interfaces/IAllocationStrategy.sol";
+import "../interfaces/IDistributionStrategy.sol";
+import "./Registry.sol";
+
 
 contract Allo is Initializable {
-    struct PoolData {
-        address identity;
-        address allocationStrategy;
-        address distributionStrategy;
+
+     error NO_ACCESS_TO_ROLE();
+
+    /// @notice Struct to hold details of an Pool
+    struct Pool {
+        bytes32 identityId;
+        IAllocationStrategy allocationStrategy;
+        IDistributionStrategy distributionStrategy;
         Metadata metadata;
         bool active;
     }
 
+    /// @notice Registry of pool creators
+    Registry public registry;
+
+    /// @notice Incremental Index
+    uint private _poolIndex;
+
+    /// @notice Pool.id -> Pool
+    mapping(uint256 => Pool) public pools;
+
     /**
      * @notice Initializes the contract after an upgrade
      * @dev In future deploys of the implementation, an higher version should be passed to reinitializer
      */
-    function initialize() public reinitializer(1) {}
+    function initialize(address _registry) public reinitializer(1) {
+        registry = Registry(_registry);
+        // ASK: should there be an update function to update registry ? Ownable contract ?
+        // Should this contract be upgradable
+    }
 
-    // Public getter on the pools mapping
-    /// @notice calls out to the registry to get the identity metadata
-    function getPoolInfo(
+    /// @notice Fetch pool and identityMetadata
+    /// @param _poolId The id of the pool
+    /// @dev calls out to the registry to get the identity metadata
+    function getIdentityInfo(
         uint256 _poolId
-    ) external view returns (PoolData memory, string memory) {
-        // Implement the function here
+    ) external view returns (Pool memory pool, string memory identityMetadata) {
+        pool = pools[_poolId];
+        identityMetadata= registry.identities(pool.identityId).metadata;
+        // ASK: why are we returning the pool
```

Comment:
@zobront

Reply:
Shouldn't we return the `identityId` so they can then look up the identity information? Does this metadata match the same addresses metadata for the same identity in the Registry contract?

```javascript
struct Identity {
  bytes32 id;
  uint index;
  string name;
  Metadata metadata;
  address anchor;
}

struct Pool {
    bytes32 identityId;
    IAllocationStrategy allocationStrategy;
    IDistributionStrategy distributionStrategy;
    Metadata metadata;
    bool active;
}
```